homeassistant/components/bmw_connected_drive/binary_sensor.py

```diff
@@ -28,38 +28,69 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+ALLOWED_CONDITION_BASED_SERVICE_KEYS = {
+    "BRAKE_FLUID",
+    "ENGINE_OIL",
+    "OIL",
+    "TIRE_WEAR_FRONT",
+    "TIRE_WEAR_REAR",
+    "VEHICLE_CHECK",
+    "VEHICLE_TUV",
+}
+
+ALLOWED_CHECK_CONTROL_MESSAGE_KEYS = {"ENGINE_OIL", "TIRE_PRESSURE"}
+
+
 def _condition_based_services(
     vehicle: MyBMWVehicle, unit_system: UnitSystem
 ) -> dict[str, Any]:
     extra_attributes = {}
     for report in vehicle.condition_based_services.messages:
+        if report.service_type not in ALLOWED_CONDITION_BASED_SERVICE_KEYS:
+            _LOGGER.warning(
+                "'%s' not an allowed condition based service (%s)",
+                report.service_type,
+                report,
+            )
+            continue
+
         extra_attributes.update(_format_cbs_report(report, unit_system))
     return extra_attributes
 
 
 def _check_control_messages(vehicle: MyBMWVehicle) -> dict[str, Any]:
     extra_attributes: dict[str, Any] = {}
     for message in vehicle.check_control_messages.messages:
-        extra_attributes.update({message.description_short: message.state.value})
+        if message.description_short not in ALLOWED_CHECK_CONTROL_MESSAGE_KEYS:
+            _LOGGER.warning(
```

Comment:
This will log every state update. We don't want that. Please just log once per entity lifetime.

Reply:
Adjusted in https://github.com/home-assistant/core/pull/76708